src/lem/tag.rs

```diff
@@ -0,0 +1,45 @@
+use std::fmt::Display;
+
+use crate::field::LurkField;
+
+#[derive(Clone, Copy, PartialEq, Debug, PartialOrd, Ord, Eq, Hash)]
+pub enum Tag {
+    // expression tags
+    Num,
+    U64,
+    Char,
+    Str,
+    Comm,
+    Fun,
+    Cons,
+    Sym,
+    Key,
+    // continuation tags
+    Outermost,
```

Comment:
Use of a flat namespace for Tags and Continuation Tags seems wrong to me — but maybe it's okay in this context. You'll need to match Lurk's pointer model when interacting with the circuit, reader, etc, though.